cpp/src/io/utilities/datasource.cpp

```diff
@@ -134,27 +134,6 @@ class file_source : public datasource {
   static constexpr size_t _gds_read_preferred_threshold = 128 << 10;  // 128KB
 };
 
-/**
- * @brief Memoized pageableMemoryAccessUsesHostPageTables device property.
- */
-[[nodiscard]] bool pageableMemoryAccessUsesHostPageTables()
-{
-  static std::unordered_map<int, bool> result_cache{};
-
-  int deviceId{};
-  CUDF_CUDA_TRY(cudaGetDevice(&deviceId));
-
-  if (result_cache.find(deviceId) == result_cache.end()) {
-    cudaDeviceProp props{};
-    CUDF_CUDA_TRY(cudaGetDeviceProperties(&props, deviceId));
-    result_cache[deviceId] = (props.pageableMemoryAccessUsesHostPageTables == 1);
-    CUDF_LOG_INFO(
-      "Device {} pageableMemoryAccessUsesHostPageTables: {}", deviceId, result_cache[deviceId]);
-  }
-
-  return result_cache[deviceId];
-}
-
 /**
  * @brief Implementation class for reading from a file using memory mapped access.
```

Comment:
We can now remove the `min_size_estimate` parameter, but I'm okay with keeping it. Let me know what you think. If we do keep it, we should probably mark it as `[[maybe_unused]]`.

Reply:
Thanks for capturing this. I've chosen to remove it.
It's a bit surprising that either the compiler (isn't there `-Wunused` ?) or linter (is there any?) didn't complain about this unused parameter. I need to have a better understanding of our build system.